schema/types/isolator.go

```diff
@@ -40,6 +44,27 @@ func AddIsolatorName(n ACIdentifier, ns map[ACIdentifier]struct{}) {
 // and PodManifest schemas.
 type Isolators []Isolator
 
+// assertValid checks that every single isolator is valid and that
+// the whole set is well built
+func (isolators Isolators) assertValid() error {
+	typesMap := make(map[ACIdentifier]uint)
+	for _, i := range isolators {
+		if err := i.Value().AssertValid(); err != nil {
+			return err
+		}
+		if typesMap[i.Name]+1 > i.Value().MaxInstances() {
+			return fmt.Errorf(`isolators set contains too many instances of type %s"`, i.Name)
```

Comment:
Thanks for the hint, I see this is part of what we were discussing today. I think it won't fit with the surrounding style here, but I'll consider it for the future.